langchain/src/chat_models/anthropic.ts

```diff
@@ -245,73 +247,107 @@ export class ChatAnthropic extends BaseChatModel implements AnthropicInput {
+    request: CompletionCreateParams & Kwargs,
     options: { signal?: AbortSignal },
     runManager?: CallbackManagerForLLMRun
-  ): Promise<CompletionResponse> {
+  ): Promise<
+    APIResponse<
+      | AnthropicApi.Completions.Completion
+      | Stream<AnthropicApi.Completions.Completion>
+    >
+  > {
     if (!this.anthropicApiKey) {
       throw new Error("Missing Anthropic API key.");
     }
-    let makeCompletionRequest;
+    let makeCompletionRequest: () => Promise<
+      APIResponse<
+        | AnthropicApi.Completions.Completion
+        | Stream<AnthropicApi.Completions.Completion>
+      >
+    >;
+
     if (request.stream) {
       if (!this.streamingClient) {
         const options = this.apiUrl ? { apiUrl: this.apiUrl } : undefined;
-        this.streamingClient = new AnthropicApi(this.anthropicApiKey, options);
+        this.streamingClient = new AnthropicApi({
+          ...options,
+          apiKey: this.anthropicApiKey,
+        });
       }
       makeCompletionRequest = async () => {
-        let currentCompletion = "";
-        return (
-          this.streamingClient
-            .completeStream(request, {
-              onUpdate: (data: CompletionResponse) => {
+        try {
+          const stream = this.streamingClient.completions.create({
+            ...request,
+          });
+
+          stream
+            .then(async (stream) => {
+              for await (const data of stream) {
+                if (options.signal?.aborted) {
+                  stream.controller.abort();
+                }
+
                 if (data.stop_reason) {
-                  return;
+                  break;
                 }
                 const part = data.completion;
                 if (part) {
-                  const delta = part.slice(currentCompletion.length);
-                  currentCompletion += delta ?? "";
                   // eslint-disable-next-line no-void
-                  void runManager?.handleLLMNewToken(delta ?? "");
+                  void runManager?.handleLLMNewToken(part ?? "");
                 }
-              },
-              signal: options.signal,
+              }
             })
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
             .catch((e: any) => {
               // Anthropic doesn't actually throw JavaScript error objects at the moment.
               // We convert the error so the async caller can recognize it correctly.
               if (e?.name === "AbortError") {
                 throw new Error(`${e.name}: ${e.message}`);
               }
               throw e;
-            })
-        );
+            });
+
+          return stream;
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (e: any) {
+          // Anthropic doesn't actually throw JavaScript error objects at the moment.
+          // We convert the error so the async caller can recognize it correctly.
+          if (e?.name === "AbortError") {
```

Comment:
I believe they do throw JS errors properly now?

Reply:
We do; AbortError's in JavaScript [weirdly don't have their own class](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort):

> Note: When abort() is called, the fetch() promise rejects with an Error of type DOMException, with name AbortError.

Reply:
Ahh wow that explains a lot. Thanks.